packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -1,7 +1,88 @@
+import isEqual from 'lodash.isequal'
+
 function applyOnBeforeDiff(before, now, fn) {
   return fn && typeof fn === 'function' ? fn(before, now) : [before, now]
 }
 
+function isDateOverlap(
+  oldValidFrom,
+  oldValidUntil,
+  newValidFrom,
+  newValidUntil
+) {
+  if (newValidFrom > oldValidFrom && newValidFrom < oldValidUntil) return true
+  if (newValidUntil < oldValidUntil && newValidUntil > oldValidFrom) return true
+  return false
+}
+
+function getPriceId(newPrice, oldVariantArray) {
+  let newPriceId
+
+  const newPriceComparison = {
+    value: { currencyCode: newPrice.value.currencyCode },
+    channel: newPrice.channel,
+    country: newPrice.country,
+    customerGroup: newPrice.customerGroup,
+  }
+
+  oldVariantArray.map(oldVariant =>
+    oldVariant.prices.find(oldPrice => {
+      const oldPriceComparison = {
+        value: {
+          currencyCode: oldPrice.value.currencyCode,
+        },
+        channel: oldPrice.channel,
+        country: oldPrice.country,
+        customerGroup: oldPrice.customerGroup,
+      }
+
+      if (
+        isEqual(
+          { ...newPriceComparison, validFrom: newPrice.validFrom },
+          { ...oldPriceComparison, validFrom: oldPrice.validFrom }
+        ) ||
+        isEqual(
+          { ...newPriceComparison, validUntil: newPrice.validUntil },
+          { ...oldPriceComparison, validUntil: oldPrice.validUntil }
+        ) ||
+        isDateOverlap(
+          oldPrice.validFrom,
+          oldPrice.validUntil,
+          newPrice.validFrom,
+          newPrice.validUntil
+        )
+      ) {
+        newPriceId = oldPrice.id
+        return true
+      }
+
+      return false
+    })
+  )
+
+  return newPriceId
+}
+
+function updateMissingPriceIds(newVariantArray, oldVariantArray) {
+  // loop over and mutate newVariant price entry
+  const addedIdsArray = []
+  newVariantArray.map(newVariant => {
+    if (!newVariant.prices) return newVariant
```

Comment:
Why do we `return` this? We never actually use it? Then we should likely return no value?